src/Logger.cc

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#include "Logger.h"
+
+#include <iostream>
+
+namespace ChimeraTK {
+
+  /********************************************************************************************************************/
+
+  Logger::~Logger() {
+    if(_mainLoopThread.joinable()) {
```

Comment:
I don't fully get this. Do you need to put the exception to the queue at all? Is boost::thread::interrupt() not sufficient to terminate the logger thread?

Reply:
No, the thread has to pass an interruption point for `boost::thread::interrupt()` to have any effect. It is technically not possible to interrupt a thread at an arbitrary position. In this case, we have to feed something to the queue so the thread wakes up and receives the interrupt signal.